mvcc/kvstore.go

```diff
@@ -160,6 +166,42 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, compactRev int64, err error) {
+	for {
+		s.mu.RLock()
+		compactMainRev := s.compactMainRev
+		currentRev := s.currentRev
```

Comment:
@heyitsanthony should HashByRev also return `currentRev` to maintenance api where it will be used as the rev in the header?